explorer/client/src/components/displaybox/DisplayBox.tsx

```diff
@@ -7,23 +7,41 @@ import {
     FALLBACK_IMAGE,
     ImageModClient,
 } from '../../utils/imageModeratorClient';
-import { transformURL } from '../../utils/stringUtils';
+import { transformURL, extractFileType } from '../../utils/stringUtils';
 
 import styles from './DisplayBox.module.css';
 
-function DisplayBox({ display }: { display: string }) {
+function DisplayBox({
+    display,
+    caption,
+}: {
+    display: string;
+    caption?: string;
+}) {
     const [hasDisplayLoaded, setHasDisplayLoaded] = useState(false);
     const [hasFailedToLoad, setHasFailedToLoad] = useState(false);
 
     const [hasImgBeenChecked, setHasImgBeenChecked] = useState(false);
     const [imgAllowState, setImgAllowState] = useState(false);
 
+    const [hasClickedImage, setHasClickedImage] = useState(false);
+
+    const [fileType, setFileType] = useState('');
+
+    useEffect(() => {
+        extractFileType(display).then((result) => setFileType(result));
+    }, [display]);
+
     const imageStyle = hasDisplayLoaded ? {} : { display: 'none' };
     const handleImageLoad = useCallback(() => {
         setHasDisplayLoaded(true);
         setHasFailedToLoad(false);
     }, [setHasDisplayLoaded]);
 
+    const handleImageClick = useCallback(() => {
+        setHasClickedImage(!hasClickedImage);
```

Comment:
You can avoid the dependency on `hasClickedImage` here (and prevent the callback function from changing) by using a set state updater:
```typescript
setHasClickedImage((prevHasClicked) => !prevHasClicked);
```